packages/trpc/server/routers/loggedInViewer/updateProfile.handler.ts

```diff
@@ -110,7 +110,7 @@ export const updateProfileHandler = async ({ ctx, input }: UpdateProfileOptions)
     // @TODO: iterate if stripeSubscriptions.hasMore is true
     const isPremiumUsernameSubscriptionActive = stripeSubscriptions.data.some(
       (subscription) =>
-        subscription.items.data[0].price.product === getPremiumPlanProductId() &&
+        subscription.items.data[0].price.id === getPremiumMonthlyPlanPriceId() &&
```

Comment:
cal.com/apps/website/pages/api/signup.ts

In signup we create the sub on the priceId and not the productId. 

So change this check to check for the right priceId